### ISSUE TITLE
gazelle not properly generating build files for `github.com/census-instrumentation/opencensus-proto/`

### ISSUE THREAD

**Issue by richardstephens** [bazelbuild/bazel-gazelle#1448]
### What version of gazelle are you using?

`v0.29.0`

### What version of rules_go are you using?

`v0.38.1`

### What version of Bazel are you using?

`v6.0.0`

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Darwin arm64

### What did you do?

Updated a fairly large go project from 1.16.2, and bumped many very out of date dependencies to their latest or close to. Most of the problems where resolved by careful selection of compatible versions, but one appears to be a bug in gazelle. For now I have forked the library and run gazelle on my fork, which works.

I have built and published a minimal reproduction case [here](https://github.com/richardstephens/gazelle-missing-dep).

### What did you expect to see?

The build should complete successfully, as it does when invoking go directly.

### What did you see instead?

```
$ bazel build //:project
INFO: Analyzed target //:project (28 packages loaded, 764 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_richard.stephens/34627ec4042a4eb707f69e3728999fc7/external/com_github_census_instrumentation_opencensus_proto/gen-go/trace/v1/BUILD.bazel:3:11: GoCompilePkg external/com_github_census_instrumentation_opencensus_proto/gen-go/trace/v1/trace.a failed: (Exit 1): builder failed: error executing command (from target @com_github_census_instrumentation_opencensus_proto//gen-go/trace/v1:trace) bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/go_sdk/builder_reset/builder compilepkg -sdk external/go_sdk -installsuffix darwin_arm64 -src ... (remaining 31 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: missing strict dependencies:
	/private/var/tmp/_bazel_richard.stephens/34627ec4042a4eb707f69e3728999fc7/sandbox/darwin-sandbox/132/execroot/__main__/external/com_github_census_instrumentation_opencensus_proto/gen-go/trace/v1/trace.pb.go: import of "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
No dependencies were provided.
Check that imports in Go sources match importpath attributes in deps.
Target //:project failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 3.075s, Critical Path: 2.47s
INFO: 29 processes: 2 internal, 27 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

### Analysis

When gazelle generates build files for the `github.com/census-instrumentation/opencensus-proto` dep, it does not include a dep for `github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1` in its build file for `github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1`.

**Comment by Geethree** [comment]
[This description holds true for me](https://github.com/bazelbuild/bazel-gazelle/issues/498#issuecomment-486387509). 

TL;DR:
Take a look at `build_extra_args` in the below go_repository rule
```starlark
    go_repository(
        name = "com_github_census_instrumentation_opencensus_proto",
        build_extra_args = ["-exclude=src"],
        importpath = "github.com/census-instrumentation/opencensus-proto",
        sum = "h1:iKLQ0xPNFxR/2hzXZMrBo8f1j86j5WHzznCCQxV/b8g=",
        version = "v0.4.1",
    )
```
